File: wiki-parser.js
/**
 Copyright 2017 KiKe. All Rights Reserved.

 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at

 http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 **/

"use strict";

const child = require('child_process').execFile;

module.exports = {

    infoBox: function (url, callback) {

        if (url == "" || url == undefined) {
            return callback(new Error("URL cannot be null"));
        }

        const type = "INFO_BOX";
        const python = child('python', [__dirname + '/wiki-parser.py', url, type]);

        let output = "";
        python.stdout.on('data', (data) => {
            output += data;
        });

        python.stderr.on('data', (data) => {
            python.kill();
            return callback(data, null);
        });

        python.on('close', (code) => {
            if (code !== 0) {
                python.kill();
                return callback(code, null);
            }
            if (code === 0) {
                return callback(null, output);
            } else {
                return callback(new Error("error in code") + code)
            }
        });
    },

    getParagraph: function (url, callback) {

        if (url == "" || url == undefined) {
            return callback(new Error("URL cannot be null"));
        }

        const type = "PARAGRAPH";
        const python = child('python', [__dirname + '/wiki-parser.py', url, type]);

        let output = "";
        python.stdout.on('data', (data) => {
            output += data;
        });

        python.stderr.on('data', (data) => {
            python.kill();
            return callback(data, null);
        });

        python.on('close', (code) => {
            if (code !== 0) {
                python.kill();
                return callback(code, null);
            }
            if (code === 0) {
                return callback(null, output);
            } else {
                return callback(new Error("error in code") + code)
            }
        });
    },

    getMainImage: function (url, callback) {

        if (url == "" || url == undefined) {
            return callback(new Error("URL cannot be null"));
        }

        const type = "IMAGE";
        const python = child('python', [__dirname + '/wiki-parser.py', url, type]);

        let output = "";
        python.stdout.on('data', (data) => {
            output += data;
        });

        python.stderr.on('data', (data) => {
            python.kill();
            return callback(data, null);
        });

        python.on('close', (code) => {
            if (code !== 0) {
                python.kill();
                return callback(code, null);
            }
            if (code === 0) {
                return callback(null, output);
            } else {
                return callback(new Error("error in code") + code)
            }
        });
    }
};

File: wiki-parser.py
import sys
import json
from wikiparser import *

try:
    url = sys.argv[1]
except IOError as err:
    print(err)

try:
    type = sys.argv[2]
except IOError as err:
    print(err)

class Path:
    def __init__(self, url):
        self._url = url
    
    def infoBox(self):
        try:
            infoBoxData =  wikiparser.infoBox(self._url)
            infoBoxData = json.dumps(infoBoxData)
            print(infoBoxData)
        except IOError as error:
            print(error)

    def getParagraph(self):
        try:
            paragraph = wikiparser.getParagraph(self._url)
            paragraph = json.dumps(paragraph)
            print(paragraph)
        except IOError as error:
            print(error)

    def getMainImage(self):
        try:
            mainImage = wikiparser.getMainImage(self._url)
            mainImage = json.dumps(mainImage)
            print(mainImage)
        except IOError as error:
            print(error)

def main():
    render = Path(url)
    if type == "INFO_BOX":
        render.infoBox()
    if type == "PARAGRAPH":
        render.getParagraph()
    if type == "IMAGE":
        render.getMainImage()


if __name__ == "__main__" : main()
